File: gamerental/src/main/java/com/example/app/user/UserId.java
package com.example.app.user;

import io.fluxzero.sdk.FluxCapacitor;
import io.fluxzero.sdk.modeling.Id;

public class UserId extends Id<UserProfile> {
    public UserId() {
        this(FluxCapacitor.generateId());
    }

    public UserId(String id) {
        super(id);
    }
}

File: flux-basic-kotlin/src/main/kotlin/com/example/app/user/api/UserId.kt
package com.example.app.user.api

import com.example.app.user.api.model.UserProfile
import io.fluxzero.sdk.Fluxzero
import io.fluxzero.sdk.modeling.Id

class UserId : Id<UserProfile> {
    constructor() : this(Fluxzero.generateId())
    
    constructor(id: String) : super(id)
}
